### ISSUE TITLE
Port more PyTorch modules

### ISSUE THREAD

**Issue by wangkuiyi** [wangkuiyi/gotorch#89]
By grepping the official DCGAN [example program](https://github.com/pytorch/examples/blob/master/dcgan/main.py), we see the following modules need to be ported before we can run DCGAN with GoTorch.

- [ ] nn.BCELoss
- [x] nn.BatchNorm2d
- [x] nn.Conv2d
- [x] nn.ConvTranspose2d
- [x] nn.LeakyReLU
- [x] nn.ReLU
- [x] nn.Sequential
- [x] nn.Sigmoid
- [x] nn.Tanh

```
$ curl -Ls https://raw.githubusercontent.com/pytorch/examples/master/dcgan/main.py | grep 'nn\.'
import torch.nn.parallel
cudnn.benchmark = True
        torch.nn.init.normal_(m.weight, 0.0, 0.02)
        torch.nn.init.normal_(m.weight, 1.0, 0.02)
        torch.nn.init.zeros_(m.bias)
class Generator(nn.Module):
        self.main = nn.Sequential(
            nn.ConvTranspose2d(     nz, ngf * 8, 4, 1, 0, bias=False),
            nn.BatchNorm2d(ngf * 8),
            nn.ReLU(True),
            nn.ConvTranspose2d(ngf * 8, ngf * 4, 4, 2, 1, bias=False),
            nn.BatchNorm2d(ngf * 4),
            nn.ReLU(True),
            nn.ConvTranspose2d(ngf * 4, ngf * 2, 4, 2, 1, bias=False),
            nn.BatchNorm2d(ngf * 2),
            nn.ReLU(True),
            nn.ConvTranspose2d(ngf * 2,     ngf, 4, 2, 1, bias=False),
            nn.BatchNorm2d(ngf),
            nn.ReLU(True),
            nn.ConvTranspose2d(    ngf,      nc, 4, 2, 1, bias=False),
            nn.Tanh()
            output = nn.parallel.data_parallel(self.main, input, range(self.ngpu))
class Discriminator(nn.Module):
        self.main = nn.Sequential(
            nn.Conv2d(nc, ndf, 4, 2, 1, bias=False),
            nn.LeakyReLU(0.2, inplace=True),
            nn.Conv2d(ndf, ndf * 2, 4, 2, 1, bias=False),
            nn.BatchNorm2d(ndf * 2),
            nn.LeakyReLU(0.2, inplace=True),
            nn.Conv2d(ndf * 2, ndf * 4, 4, 2, 1, bias=False),
            nn.BatchNorm2d(ndf * 4),
            nn.LeakyReLU(0.2, inplace=True),
            nn.Conv2d(ndf * 4, ndf * 8, 4, 2, 1, bias=False),
            nn.BatchNorm2d(ndf * 8),
            nn.LeakyReLU(0.2, inplace=True),
            nn.Conv2d(ndf * 8, 1, 4, 1, 0, bias=False),
            nn.Sigmoid()
            output = nn.parallel.data_parallel(self.main, input, range(self.ngpu))
criterion = nn.BCELoss()
```

**Comment by wangkuiyi** [comment]
We now have three runnable examples: MNIST, DCGAN, and ResNet50. We will reopen issues for automatically migrate PyTorch functionalities.
